app/models/miq_server/role_management.rb

```diff
@@ -239,6 +239,13 @@ def synchronize_active_roles(servers, roles_to_sync)
         elsif delta > 0
           delta.times do
             if inactive.length > 0
+              # We need to sort by:
+              #   * priority (some roles have 'affinity' to run on a specific server)
+              #   * by the number of active roles on a server
+              inactive = inactive.sort_by do |server, priority|
+                [priority, server.reload.active_roles.size]
+              end
+
```

Comment:
Note, we probably have to do something similar on line 251/258 below when it migrates roles from one server to another.  It needs to choose the destination server based on priority (affinity) AND the number of roles active on that server.

👇